src/riak_kv_vnode.erl

```diff
@@ -2383,12 +2383,29 @@ encode_and_put_no_sib_check(Obj, Mod, Bucket, Key, IndexSpecs, ModState,
                         false ->
                             ok
                     end,
-                    PutRet = Mod:put(Bucket, Key, IndexSpecs, EncodedVal,
-                                     ModState),
+                    PutRet = case timeseries_data(Key, Obj) of
+                                 true ->
+                                     RawKey = sext:decode(Key),
+                                     Mod:put(Bucket, RawKey, IndexSpecs, EncodedVal,
+                                             ModState);
+                                 false ->
+                                     Mod:put(Bucket, Key, IndexSpecs, EncodedVal,
+                                             ModState)
+                             end,
                     {PutRet, EncodedVal}
             end
     end.
 
+%% Look for sext-encoded tuple (key) because handoff lifts it straight out of
+%% the backend and gives it to us as is. If this isn't sext-encoded
+%% then this ain't timeseries so we don't need to dig through the
+%% object metadata for a DDL.
+timeseries_data(<<16,0,0,0, _Rest/binary>>, Object) ->
+    {IsTs, _Ver} = riak_object:is_ts(Object),
```

Comment:
The comment above says "we don't need to dig ... for a DDL", but riak_object:is_ts/1 does just that. The cost of "digging" pales in comparison to losing data.

If we are 100% certain that only TS data is handed off as is, then the comment should remain and the verification that the object contains TS data should be removed. Otherwise, the comment should be removed and the verification should remain.

Besides the comparison between plain-old KV objects, we should consider CRDT objects.

I recommend we err on the side of preserving data, but jump on testing the various object types. We can more readily test the performance impact of digging into the riak_object for DDL on handoff.